lib/trino-filesystem-s3/src/main/java/io/trino/filesystem/s3/S3SecurityMapping.java

```diff
@@ -86,6 +90,10 @@ public S3SecurityMapping(
 
         this.allowedKmsKeyIds = ImmutableSet.copyOf(allowedKmsKeyIds.orElse(ImmutableList.of()));
 
+        this.customerKey = requireNonNull(customerKey, "customerKey is null");
+
+        this.allowedCustomerKeys = ImmutableSet.copyOf(allowedCustomerKeys.orElse(ImmutableList.of()));
```

Comment:
this can be: `allowedCustomerKeys.map(ImmutableSet::copyOf).orElse(ImmutableList.of())`